sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -203,6 +204,62 @@ case class GpuTimeSub(start: Expression,
   }
 }
 
+case class GpuDateAddInterval(start: Expression,
+    interval: Expression,
+    timeZoneId: Option[String] = None)
+    extends GpuTimeMath(start, interval, timeZoneId) {
+
+  override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression = {
+    copy(timeZoneId = Option(timeZoneId))
+  }
+
+  override def intervalMath(us_s: Scalar, us: ColumnVector): ColumnVector = {
+    us.add(us_s)
+  }
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(DateType, CalendarIntervalType)
+
+  override def dataType: DataType = DateType
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+
+    withResourceIfAllowed(left.columnarEval(batch)) { lhs =>
+      withResourceIfAllowed(right.columnarEval(batch)) { rhs =>
+        (lhs, rhs) match {
+          case (l: GpuColumnVector, intvl: CalendarInterval) =>
+            if (intvl.months != 0) {
+              throw new UnsupportedOperationException("Months aren't supported at the moment")
+            }
+            val microSecondsInOneDay = TimeUnit.DAYS.toMicros(1)
+            val microSecToDays = if (intvl.microseconds < 0) {
+              // This is to calculate when subtraction is performed. Need to take into account the
+              // interval( which are less than days). Convert it into days which needs to be
+              // subtracted along with intvl.days(if provided).
+              (intvl.microseconds.abs.toDouble / microSecondsInOneDay).ceil.toInt * -1
+            } else {
+              (intvl.microseconds.toDouble / microSecondsInOneDay).toInt
+            }
+            val daysToAdd = intvl.days + microSecToDays
+            if (daysToAdd != 0) {
+              withResource(Scalar.fromInt(daysToAdd)) { us_s =>
```

Comment:
I hadn't tried DURATION_DAYS earlier but I am getting below exception:
```
  : ai.rapids.cudf.CudfException: NVRTC error: NVRTC_ERROR_COMPILATION
	at ai.rapids.cudf.ColumnView.binaryOpVS(Native Method)
	at ai.rapids.cudf.ColumnView.binaryOp(ColumnView.java:907)
	at ai.rapids.cudf.ColumnView.binaryOp(ColumnView.java:897)```
Have updated the code to use `logicalCastTo` instead of `castTo` . Please take another look.

Reply:
Could you try to clean your cudf jit cache and try again? If you still at least check with cudf if that is expected. I don't think it is and we might need to file a bug with them.